### DISCUSSION TITLE
Migration failed on django.db.migrations.exceptions.NodeNotFoundError:

### DISCUSSION THREAD

**Discussion by Virgule59** [netbox-community/netbox#7955] [Q&A] [answered]
Hello, 

I try to migrate on new server and new installation of netbox 3.0.11 on ubuntu 20.04.3, my old version of netbox 2.6 by migrate database.
On the new server no problem to access netbox webpage.

After installation of netbox 
I migrated database via pg_dump and restore the database on my new server but when i launch upgrade.sh, i have an error. and script failed on this message.

`django.db.migrations.exceptions.NodeNotFoundError: Migration dcim.0131_consoleport_speed depends on nonexistent node ('dcim', '0130_sitegroup'). Django tried to replace migration dcim.0130_sitegroup with any of [dcim.0003_squashed_0130] but wasn't able to because some of the replaced migrations are already applied.`

Can you help me please ?

**Comment by ziggekatten** [comment, marked as answer]
You can't upgrade from 2.6 directly. 

You need to go to 2.9.x if I remember correctly before going 3.x. You can't go directly to 3.x. 

But, the good news, it does work! We have been doing upgrades since before 2.6 with no hickups. We have however followed the versions a little bit more active than you have:-)

**Reply by Virgule59** [reply]
Thanks you for tour advice i will try you that and do a reply if that work !

**Reply by Virgule59** [reply]
Hi Ziggekatten,
installation of netbox 3.0.11 work fine. Ive just add the version 2.11.12 in the upgrade path which from 2.6 give 2.6->2.9.11->2.11.12->3.0.11
Best Regards

**Reply by candlerb** [reply]
You can go 2.6.x -> 2.11.x -> 3.0.x

**Reply by rvalenciath** [reply]
Hi team,

Why you don't add this information into the How to Upgrade Netbox official link (https://netbox.readthedocs.io/en/stable/installation/upgrading/)? It will be really helpful for all of us upgrading from 2.x to 3.x.

**Reply by rvalenciath** [reply]
Just in case, I did the path following your recommendation: 2.10.2 > 2.11.12 > 3.0.12 > 3.1.8 (last one 17th Feb. 2022)

Please, add path into: Upgrading to a New NetBox Release (https://netbox.readthedocs.io/en/stable/installation/upgrading/)

Thanks.
